packages/next-app/src/hooks/useTransaction.ts

```diff
@@ -274,36 +275,75 @@ export function useInternalBalancesTransaction({
     args: [userBalancesOp],
   });
 
-  const { data, write } = useVaultManageUserBalance(config);
+  const { data, write, error } = useVaultManageUserBalance(config);
 
-  //function to prepare depoist transaction
-  async function approveToken() {
-    if (!submitData) return;
-    setTransactionStatus(TransactionStatus.WAITING_APPROVAL);
-    setNotification(
-      NOTIFICATION_MAP_INTERNAL_BALANCES[TransactionStatus.WAITING_APPROVAL]
-    );
-    const config = await prepareWriteContract({
-      address: submitData[0].tokenAddress as Address,
+  //check allowance
+  async function checkAllowance({
+    tokenAmount,
+    tokenAddress,
+    tokenDecimals,
+  }: {
+    tokenAmount: string;
+    tokenAddress: Address;
+    tokenDecimals: number;
+  }) {
+    if (tokenAmount === "" || Number(tokenAmount) <= 0) {
+      setHasAllowance(undefined); // User doesn't have enough allowance
+      return;
+    }
+    const allowance = await readContract({
+      address: tokenAddress,
       abi: erc20ABI,
-      functionName: "approve",
-      args: [
-        vaultAddress[5],
-        parseFixed(
-          submitData[0]?.tokenAmount ? submitData[0].tokenAmount : "0",
-          submitData[0]?.tokenDecimals ? submitData[0].tokenDecimals : "0"
-        ),
-      ],
+      functionName: "allowance",
+      args: [userAddress, vaultAddress[5]],
     });
-    const data = await writeContract(config);
-    const { hash, wait } = data;
-    handleTransactionStatus({ hash });
-    const receipt = await wait();
-    if (receipt.status) {
+    const amountToApprove = parseFixed(tokenAmount, tokenDecimals);
+    if (allowance.gte(amountToApprove)) {
+      setHasAllowance(true); // User has enough allowance
```

Comment:
if you re-name `hasAllowance` to be more explicit maybe the comment isn't even needed, some examples: `isApprovalRequired` `hasEnoughAllowance` etc